adal/versioning/app_version.gradle

```diff
@@ -0,0 +1,123 @@
+def getVersionFile() {
+    return file('versioning/version.properties')
+}
+
+def getVersionProps() {
+    def versionPropsFile = getVersionFile()
+    def versionProps = new Properties()
+    versionProps.load(new FileInputStream(versionPropsFile))
+    return versionProps
+}
+
+private Integer getVersionNamePatch() {
+    return (getVersionProps()['versionName'] =~ /\d+/)[2].toInteger()
+}
+
+private Integer getVersionNameMinor() {
+    return (getVersionProps()['versionName'] =~ /\d+/)[1].toInteger()
+}
+
+private Integer getVersionNameMajor() {
+    return (getVersionProps()['versionName'] =~ /\d+/)[0].toInteger()
+}
+
+private Integer getVersionNightly(){
+    return getVersionProps()['versionNightly'].toInteger()
+}
+
+private Integer getVersionRC(){
+    return getVersionProps()['versionRC'].toInteger()
+}
+
+private void commit(String versionName, String task) {
+    Process addChanges = ['git', 'add', 'adal/versioning/version.properties'].execute(null, project.rootDir)
+    addChanges.waitForProcessOutput(System.out, System.err)
+
+    Process createCommit = ['git', 'commit', "-m ${task} Release ${versionName}"].execute(null,  project.rootDir)
+    createCommit.waitForProcessOutput(System.out, System.err)
+}
+
+ext.getAppVersionCode = {
+    getVersionProps()['versionCode'].toInteger()
```

Comment:
I assume version props is actually going to be ahead of whatever the last build was.